src/System.Private.CoreLib/shared/System/Globalization/TextInfo.cs

```diff
@@ -150,43 +124,28 @@ internal void SetReadOnlyState(bool readOnly)
         }
 
 
-        ////////////////////////////////////////////////////////////////////////
-        //
-        //  ListSeparator
-        //
-        //  Returns the string used to separate items in a list.
-        //
-        ////////////////////////////////////////////////////////////////////////
+        /// <summary>
+        /// Returns the string used to separate items in a list.
+        /// </summary>
         public virtual string ListSeparator
         {
-            get
-            {
-                if (_listSeparator == null)
-                {
-                    _listSeparator = _cultureData.SLIST;
-                }
-                return _listSeparator;
-            }
-
+            get => _listSeparator ?? _cultureData.ListSeparator;
```

Comment:
>_cultureData.ListSeparator; [](start = 37, length = 27)

you need to initialize _listSeparator if it is null

Reply:
Fixed